yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1179,8 +1182,35 @@ object Client extends Logging {
    * Add the given path to the classpath entry of the given environment map.
    * If the classpath is already set, this appends the new path to the existing classpath.
    */
-  private def addClasspathEntry(path: String, env: HashMap[String, String]): Unit =
-    YarnSparkHadoopUtil.addPathToEnvironment(env, Environment.CLASSPATH.name, path)
+  private def addClasspathEntry(
+      conf: SparkConf,
+      path: String,
+      env: HashMap[String, String]): Unit =
+    YarnSparkHadoopUtil.addPathToEnvironment(env, Environment.CLASSPATH.name,
+      getClusterPath(conf, path))
```

Comment:
I think it should be safe, because it's very unlikely that those paths would contain the string being replaced, but I see your point. Let me take a look.